### DISCUSSION TITLE
How to determinate if null has been set for some field explicitly or hasn't been entered at all

### DISCUSSION THREAD

**Discussion by VitalijF** [Netflix/dgs-framework#796] [General]
For example we have mutation:

`doSomething(input: TestInput!) : SomeResponse`

with next input:

```
input TestInput {
  mandatoryField: String!
  optionalField: String
}
```

Is it any way to know is null has been explicitly set to **optionalField** or hasn't been entered at all

Example with explicitly _null_ 
```
mutation {
    doSomething(input: {
        mandatoryField: "Test"
        optionalField: null
    }) {
        response 
    }
}
```

Example without explicitly _null_
```
mutation {
    doSomething(input: {
        mandatoryField: "Test"
    }) {
        response 
    }
}
```

**Comment by srinivasankavitha** [comment]
You can check whether that exists in your data fetcher using the Data Fetching Environment. The DFE gives you access to the input arguments that are passed in and only the fields that are set would be available there. Note that if you deserialize the input arguments into corresponding POJOs, you could lose that information since the optional fields would be automatically set to null.

```
public String helloDataFetchingEnvironment environment) {
        Map<String, ?> testInput = environment.getArgument("input")
        // check if testInput has "optionalField"
       if (testInput.get("optionalField") {
            // your logic
       }
    }
```
